src/In.ProjectEKA.HipService/DataFlow/DataFlow.cs

```diff
@@ -6,51 +6,97 @@ namespace In.ProjectEKA.HipService.DataFlow
-                        ErrorMessage.ContextArtefactIdNotFound)));
-            }
+            if (consent == null) return ConsentArtefactNotFound();
 
             var dataRequest = new DataRequest(consent.ConsentArtefact.CareContexts, request.HiDataRange,
                 request.CallBackUrl, consent.ConsentArtefact.HiTypes, request.TransactionId);
-            var result = await dataFlowRepository.SaveRequestFor(request.TransactionId, request)
+            var result = await dataFlowRepository.SaveRequest(request.TransactionId, request)
                 .ConfigureAwait(false);
             var (response, errorRepresentation) = result.Map(r =>
             {
                 var errorResponse = new ErrorRepresentation(new Error(ErrorCode.ServerInternalError,
                     ErrorMessage.InternalServerError));
-                return new Tuple<HealthInformationResponse, ErrorRepresentation>(null, errorResponse);
-            }).ValueOr(new Tuple<HealthInformationResponse,
-                ErrorRepresentation>(new HealthInformationResponse(request.TransactionId), null));
+                return new Tuple<HealthInformationTransactionResponse, ErrorRepresentation>(null, errorResponse);
+            }).ValueOr(new Tuple<HealthInformationTransactionResponse,
+                ErrorRepresentation>(new HealthInformationTransactionResponse(request.TransactionId), null));
 
             if (errorRepresentation == null)
             {
                 PublishDataRequest(dataRequest);
             }
 
-            return new Tuple<HealthInformationResponse, ErrorRepresentation>(response, errorRepresentation);
+            return new Tuple<HealthInformationTransactionResponse, ErrorRepresentation>(response, errorRepresentation);
+        }
+
+        private static Tuple<HealthInformationTransactionResponse, ErrorRepresentation> ConsentArtefactNotFound()
+        {
+            return new Tuple<HealthInformationTransactionResponse, ErrorRepresentation>(null,
+                new ErrorRepresentation(new Error(ErrorCode.ContextArtefactIdNotFound,
+                    ErrorMessage.ContextArtefactIdNotFound)));
+        }
+
+        public async Task<Tuple<HealthInformationResponse, ErrorRepresentation>> HealthInformationFor(
+            string informationId,
+            string token,
+            string transactionId)
+        {
+            var healthInformation = await healthInformationRepository.GetAsync(informationId);
+            return healthInformation
+                .Map(information => HealthInformation(token, transactionId, information))
+                .ValueOr(ErrorOf(ErrorResponse.HealthInformationNotFound));
+        }
+
+        private Tuple<HealthInformationResponse, ErrorRepresentation> HealthInformation(
+            string token,
+            string transactionId,
+            HealthInformation information)
+        {
+            if (information.Token != token) return ErrorOf(ErrorResponse.InvalidToken);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5067092)